pkg/antctl/raw/packetcapture/command.go

```diff
@@ -0,0 +1,327 @@
+		} else if len(kv) == 1 {
+			if len(kv[0]) != 0 {
+				fields[v] = 0
+			}
+		} else {
+			return nil, fmt.Errorf("%s is not valid in flow", v)
+		}
+	}
+	return fields, nil
+}
+
+func getPodFile(cmd *cobra.Command) (PodFileCopy, error) {
+	config, client, _, err := getClients(cmd)
+	if err != nil {
+		return nil, err
+	}
+	return &podFile{
+		restConfig: config,
+		client:     client,
+	}, nil
+}
+
+func getConfigAndClients(cmd *cobra.Command) (*rest.Config, kubernetes.Interface, antrea.Interface, error) {
+	kubeConfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	k8sClientset, client, err := raw.SetupClients(kubeConfig)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to create clientset: %w", err)
+	}
+	return kubeConfig, k8sClientset, client, nil
+}
+
+func getPCName(src, dest string) string {
+	replace := func(s string) string {
+		return strings.ReplaceAll(s, "/", "-")
+	}
+	prefix := fmt.Sprintf("%s-%s", replace(src), replace(dest))
+	if option.nowait {
+		return prefix
+	}
+	return fmt.Sprintf("%s-%s", prefix, rand.String(8))
+}
+
+func packetCaptureRunE(cmd *cobra.Command, args []string) error {
+	option.timeout, _ = cmd.Flags().GetDuration("timeout")
+	if option.timeout > 300*time.Second {
+		return errors.New("timeout cannot be longer than 1 hour")
+	}
+	if option.timeout == 0 {
+		option.timeout = defaultTimeout
+	}
+	if option.number == 0 {
+		return errors.New("packet number should be larger than 0")
+	}
+
+	_, _, antreaClient, err := getClients(cmd)
+	if err != nil {
+		return err
+	}
+	pc, err := newPacketCapture()
+	if err != nil {
+		return fmt.Errorf("error when constructing a PacketCapture CR: %w", err)
+	}
+	createCtx, cancel := context.WithTimeout(cmd.Context(), 5*time.Second)
+	defer cancel()
+
+	if _, err := antreaClient.CrdV1alpha1().PacketCaptures().Create(createCtx, pc, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error when creating PacketCapture, is PacketCapture feature gate enabled? %w", err)
+	}
+	defer func() {
+		if !option.nowait {
+			if err = antreaClient.CrdV1alpha1().PacketCaptures().Delete(context.TODO(), pc.Name, metav1.DeleteOptions{}); err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "error when deleting PacketCapture: %s", err.Error())
+			}
+		}
+	}()
+
+	if option.nowait {
+		fmt.Fprintf(cmd.OutOrStdout(), "PacketCapture Name: %s\n", pc.Name)
+		return nil
+	}
+
+	var latestPC *v1alpha1.PacketCapture
+	err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, option.timeout, false, func(ctx context.Context) (bool, error) {
+		res, err := antreaClient.CrdV1alpha1().PacketCaptures().Get(ctx, pc.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range res.Status.Conditions {
+			if cond.Type == v1alpha1.PacketCaptureComplete && cond.Status == metav1.ConditionTrue {
+				latestPC = res
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	if wait.Interrupted(err) {
```

Comment:
my concern is that when the timeout is hit, these 2 events will happen at roughly the same time:
1) the PacketCapture expires and the agent writes the partial pcap to disk
2) the `wait.PollUntilContextTimeout` is interrupted and the antctl command exits right away, without retrieving the capture results

Do you think this can happen?
Maybe after the `wait.PollUntilContextTimeout` expires, we could wait for the results for an extra (at most) 5 seconds. What do you think?

Reply:
the concern make sense.updated with increased timeout for `PollUntilContextTimeout`. If the PC timeout, and we have a pacap file path, it will be copied

Reply:
lgtm, can you add a comment before the `wait.PollUntilContextTimeout` line to explain why we are doing this